cmake/mission_defaults.cmake

```diff
@@ -0,0 +1,24 @@
+##################################################################
```

Comment:
Well, it kinda defeats the purpose of it -- if the "defaults" are mission defined just like the targets.cmake file is, then just put it in targets.cmake.  There is no need for a separate file at that point if they reside in the same scope. 

It is less flexible/robust and not backward compatible - a user must modify their config just to get the defaults, and we'll have similar backward incompatibilities every time we need to add/remove/modify something in the default set of modules.  

It also means we have no common way to handle stuff such as extra libraries when ENABLE_UNIT_TESTS is true.  Every user will have to add this on their own.

One of the problems with classic build is that it had a whole lot of logic as "clone and own" files which were still tightly coupled with the distribution files, and this resulted in frequent build breakage for users following the master branch.  The more logic we force users to clone and own, the more likely that it is going to break with future changes requiring manual update.  This is fine for power users who have opted in and know they are overriding some portion of the cfe, but not good for normal users who just want something that works out of the box.

Reply:
Sure, valuable part of the trade... but on the other side there's quite a bit of "magic" scattered about the build system.  Without digging into the system to figure out what the various search paths are, what directories are included, what objects are linked at which level, etc it's a challenge for a normal user to trace down everything going on.

Perhaps improved reporting and documentation would be a good approach?

Reply:
Possible compromise in commit 6d64be0.... this puts a `MISSION_MODULE_SEARCH_PATH` into the defaults file for more visibility but keeps it in the `cfe/cmake` directory to reduce the number of things the user needs to clone and own.

Reply:
Yup, good update.  I'd also think moving the hard coded include directories to mission defaults would make sense.  What about just defining MISSION_DEPS there also as the full list?

Reply:
`MISSION_DEPS` is a derived/calculated value, based on the combined set of modules defined in all the various TGT lists in targets.cmake.  The user should not directly set this.

Currently there are two implicit/fixed entries in this list - `cfe-core` and `osal` -- which we could move these to a separate list in the defaults file.  However these targets are directly referenced in other parts of the build too, so any change to these two would likely break other parts of the build.  But if we use interface libraries then maybe that breakage could be avoided too (i.e. we could theoretically allow a user to link with a complete alternative osal that isn't called osal).

Reply:
Actually, looking at the code again, it is `MISSION_APPS` that is derived, and `MISSION_DEPS` is really just `MISSION_CORE_MODULES` with the two fixed entities added on.  Maybe we can just use `MISSION_CORE_MODULES` instead, I will try this.

Reply:
OK - regarding previous comment I think its possible to consolidate these two implicit deps into the `MISSION_CORE_MODULES` list, but it depends on getting PR #728 merged, because of the way the target names are handled.  In that PR things are simplified so that the targets are actually named "cfe-core" and "osal" respectively so it just works.  But because the target is named "cfe-core_default_cpu1" this still needs special handling.
I will submit a new issue about this which I can implement after the PR #728 stuff is settled.

Reply:
See #746